tests/testSBMLSuite.py

```diff
@@ -165,18 +168,17 @@ def concentrations_to_amounts(
 ):
     """Convert AMICI simulated concentrations to amounts"""
     for species in amount_species:
-        # Skip "species" that are actually compartments
-        compartment_species = [str(c) for c in wrapper.compartment_symbols] + \
-            list(set([
-                str(s) for s in wrapper.species_rate_rules
-                if wrapper.species_has_only_substance_units[
-                    wrapper.species_index[str(s)]
-                ]
-            ]).difference(requested_concentrations))
-        if not species == '' and species not in compartment_species:
-            symvolume = wrapper.species_compartment[
-                wrapper.species_index[species]
-            ]
+        # Skip "species" that are actually compartments or rate rules that
+        # already specify amounts
+        amt_species = list(set(
+            str(sid) for sid, s in wrapper.symbols[SymbolId.SPECIES].items()
+            if s['amount']
+        ).difference(requested_concentrations))
```

Comment:
Again, not sure if anything specific to rate rules is here in code, although rate rules is mentioned in the comment.

Reply:
👍